lib/charms/opensearch/v0/opensearch_plugin_manager.py

```diff
@@ -76,19 +88,37 @@ def plugins(self) -> List[OpenSearchPlugin]:
             plugins_list.append(new_plugin)
         return plugins_list
 
+    def get_plugin(self, plugin_class: OpenSearchPlugin) -> OpenSearchPlugin:
+        """Returns a given plugin based on its class."""
+        for plugin in self.plugins:
+            if isinstance(plugin, plugin_class):
+                return plugin
+        raise KeyError(f"Plugin manager did not find plugin: {plugin_class}")
+
     def _extra_conf(self, plugin_data: Dict[str, Any]) -> Optional[Dict[str, Any]]:
         """Returns the config from the relation data of the target plugin if applies."""
         relation_name = plugin_data.get("relation")
         relation = self._charm.model.get_relation(relation_name) if relation_name else None
         if not relation:
             return None
-        return relation.data[self._charm.app]
+        app = self._charm.model.get_relation(relation_name).app
+        return relation.data[app]
 
     def run(self) -> bool:
         """Runs a check on each plugin: install, execute config changes or remove.
 
         This method should be called at config-changed event. Returns if needed restart.
         """
+        if (
+            not self._charm.opensearch.is_started()
+            or self._charm.health.apply() != HealthColors.GREEN
+        ):
+            # If the health is not green, then raise a cluster-not-ready error
```

Comment:
What is the reasoning behind this?
A cluster can be yellow when it doesn't have enough nodes to satisfy the replication constraints set on an index, or when shards are moving around.

Reply:
Added the yellow as well